### PR TITLE
Refactor and updating of structure and overall capabilities

### PR DESCRIPTION
**Pull request by jameswburke**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,60 @@
+sudo: false
```
Comment (renatonascalves) [inline .travis.yml]: This is not needed anymore!

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+dist: trusty
```
Comment (renatonascalves) [inline .travis.yml]: Neither is this!

File: inc/objects/class-term.php [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		add_action( "{$taxonomy}_edit_form", [ $this, 'add_meta_boxes' ], 1000 );
+	}
+
+	/**
+	 * Render a table of post meta.
+	 */
+	public function add_meta_boxes() {
+
+		// Ensure the term_id is set.
+		if ( ! isset( $_GET['tag_ID'] ) ) {
```
Comment (renatonascalves) [inline inc/objects/class-term.php]: I'm not 100% sure this will work for other taxonomies. Is this only for the tag taxonomy?

Comment (jameswburke) [reply inc/objects/class-term.php]: Works for them all.

**Pull request merged**